### ISSUE TITLE
runPythonAsync returns a Promise but annotated with Py2JsResult type

### ISSUE THREAD

**Issue by LinHeLurking** [pyodide/pyodide#1926]
I'm trying typescript by adding those `*.d.ts` files in Github release into `node_modules/pyodide/`. But I found `runPythonAsync` has incorrect type hint.

Here's the sample codes.

```TypeScript
export async function test_pyodide() {
    let pyodide_pkg = await import("pyodide/pyodide");

    let pyodide_promise = pyodide_pkg.loadPyodide({
        indexURL: "/pyodide_full",
    });

    pyodide_promise.then(py => {
        console.log("Python loaded!");

        const res_promise = py.runPythonAsync("1+1");
        // @ts-ignore
        res_promise.then(res => {
            console.log(res);
        });

        console.log(py.runPythonAsync("1+1"));
    });
}
```

Then in browser console there are these logs:

```
Promise
  [[Prototype]]: Promise
  [[PromiseState]]: "fulfilled"
  [[PromiseResult]]: 2

2
```

**Comment by hoodmane** [comment]
Thanks for the report! 

This is the result of an expectation that `async` functions would automatically have their return type wrapped in `Promise` but this is apparently not the case. We really need to switch to using typescript rather than jsdoc types -- I think typescript does this?

**Comment by hoodmane** [comment]
Resolved by #2133.
